rpc/handler.go

```diff
@@ -390,7 +390,14 @@ func (h *handler) handleSubscribe(cp *callProc, msg *jsonrpcMessage) *jsonrpcMes
 func (h *handler) runMethod(ctx context.Context, msg *jsonrpcMessage, callb *callback, args []reflect.Value) *jsonrpcMessage {
 	result, err := callb.call(ctx, msg.Method, args)
 	if err != nil {
-		return msg.errorResponse(err)
+		if ec, ok := err.(Error); ok {
+			return msg.errorResponse(ec)
+		} else {
+			return msg.errorResponse(&internalServerError{
+				code: errorcodeInternalServerError,
+				message: err.Error(),
+			})
+		}
```

Comment:
This change is not necessary, why did you reintroduce it? If the method implementation returns an error, it is not an 'internal error', it's just a normal error. We can simply use the default error code of -32000 in this situation.

Reply:
Let me clarify it even more. The definition of 'internal error' is: something failed in the processing of the request. The **result** of processing can also be an error. But such errors are not considered 'internal'.

Reply:
makes sense. I reverted those changes. Thanks for reviewing!